torch/serialization.py

```diff
@@ -332,10 +332,23 @@ def __init__(self, name_or_buffer) -> None:
 
 class _open_zipfile_writer_file(_opener):
     def __init__(self, name) -> None:
-        super().__init__(torch._C.PyTorchFileWriter(str(name)))
+        self.file_stream = None
+        self.name = str(name)
+        try:
+            self.name.encode('ascii')
+        except UnicodeEncodeError:
+            # PyTorchFileWriter only supports ascii filename.
+            # For filenames with non-ascii characters, we rely on Python
+            # for writing out the file.
+            self.file_stream = io.FileIO(self.name, mode='w')
```

Comment:
`mode=w` is text mode, it may corrupt binary files on Windows due to newline code conversion.
I think this should be `mode='wb'`.

Reply:
I think the `mode` option is confusing here but should be doing the right thing:

```python
>>> import io
>>> f = io.FileIO("tmp", 'w')
>>> f
<_io.FileIO name='tmp' mode='wb' closefd=True>
```

Note that the internal `mode` of the FileIO object is `wb`, different from the user-specified mode of `w`.

Separately, [The `writelines()` method](https://docs.python.org/3/library/io.html#io.IOBase.writelines) that FileIO inherits from BaseIO does not add any newlines per the linked documentation.

Reply:
Sorry, I was confusing it with `open` arguments. You are right.